src/libp2p.js

```diff
@@ -240,6 +240,21 @@ class DaemonLibp2p extends Libp2p {
             this.peerInfo.multiaddrs.add(addr)
           })
         }
+
+        // temporary removal of "/ipfs/..." from multiaddrs
+        // this will be solved in: https://github.com/libp2p/js-libp2p/issues/323
+        const finalMultiaddrs = this.peerInfo.multiaddrs.toArray().map(m => {
+          let ma
+          try {
+            ma = m.decapsulate('ipfs')
+          } catch (_) {
+            ma = m
+          }
+          return ma
```

Comment:
It would be good to replace `map` with `forEach` and do `this.peerInfo.multiaddrs.replace(m, ma)` here to avoid the additional iteration.